cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go

```diff
@@ -19,127 +19,255 @@ limitations under the License.
+//
+// Basics
+//
+// The preferred way to configure kubeadm is to pass YAML a config file with the --config option: some of the
+// configuration options defined in the kubeadm config file are also available as command line flags, but only
+// the most common/simple use case are supported with this approach.
+//
+// A kubeadm config file could contain multiple configuration types separated using three dashes (“---”).
+//
+// The kubeadm config print-defaults command print the default values for all the kubeadm supported configuration types.
+//
+//     apiVersion: kubeadm.k8s.io/v1alpha3
+//     kind: InitConfiguration
+//         ...
+//     ---
+//     apiVersion: kubeadm.k8s.io/v1alpha3
+//     kind: ClusterConfiguration
+//         ...
+//     ---
+//    apiVersion: kubelet.config.k8s.io/v1beta1
+//    kind: KubeletConfiguration
+//         ...
+//     ...
+//     ---
+//    apiVersion: kubeproxy.config.k8s.io/v1alpha1
+//    kind: KubeProxyConfiguration
+//         ...
+//     ...
+//     ---
+//    apiVersion: kubeadm.k8s.io/v1alpha3
+//    kind: JoinConfiguration
+//         ...
+//     ...
+//
+// The list of configuration types that must be included in a configuration file depends by the action you are
+// performing (init or join) and by the configuration options you are going to use (defaults or advanced customization).
+//
+// If some configuration types are not provided, or provided only partially, kubeadm will use default values; defaults
+// provided by kubeadm includes also enforcing consistency of values across components when required (e.g.
+// cluster-cidr flag on controller manager and clusterCIDR on kube-proxy).
+//
+// User are always allowed to override default values, with the only exception of a small subset of setting with
+// relevance for security (e.g. enforce authorization-mode Node and RBAC on api server)
+//
+// Starting from v1.12.1, if the user provides a configuration types that is not expected for the action you are performing,
+// kubeadm will ignore those types.
+//
+// Kubeadm init configuration types
+//
+// When executing kubeadm init with the --config option, following configuration types could be used, but
+// only one between InitConfiguration and ClusterConfiguration is mandatory.
+//
+//     apiVersion: kubeadm.k8s.io/v1alpha3
+//     kind: InitConfiguration
+//     bootstrapTokens:
+//         ...
+//     nodeRegistration:
+//         ...
+//     apiEndpoint:
+//         ...
+//
+// InitConfiguration (and as well ClusterConfiguration afterwards) are originated from MasterConfiguration type
+// in v1alpha2 kubeadm config version.
+//
+// The InitConfiguration type should be used to configure runtime settings, that in case of kubeadm init
+// are the configuration of the bootstrap token and all the setting which are specific to the node where kubeadm
+// is executed, including:
+//
+// NodeRegistration, that holds fields that relate to registering the new node to the cluster;
```

Comment:
we should probably indicate that NodeRegistration and APIEndpoint are bullet items of the above `, including:` list?

something like
```
is executed, including:
  * NodeRegistration .....
  * APIEndpoint ....
```